app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,56 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.dto.UserLoginData;
+import com.codesoom.assignment.errors.InvalidTokenException;
+import com.codesoom.assignment.errors.UserNotFoundException;
+import com.codesoom.assignment.errors.WrongPasswordException;
+import com.codesoom.assignment.utils.JwtUtil;
+import com.github.dozermapper.core.Mapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private final UserService userService;
+    private final Mapper mapper;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserService userService , Mapper dozerMapper) {
+        this.jwtUtil = jwtUtil;
+        this.userService = userService;
+        this.mapper = dozerMapper;
+
+    }
+
+    /**
+     * JWT 토큰을 반환한다.
+     *
+     * @param loginData 로그인 정보
+     * @throws UserNotFoundException 로그인 정보에 해당하는 사용자가 존재하지 않을 경우
+     * @throws WrongPasswordException 사용자의 패스워드 정보가 일치하지 않은 경우
+     * @return JWT 반환
+     */
+    public String login(UserLoginData loginData){
+        User user = mapper.map(loginData , User.class);
+        User findUser = userService.findByEmail(user.getEmail());
```

Comment:
```suggestion
        User findUser = userService.findByEmail(emailFrom(loginData));
```

이런식으로 메소드를 추출해본다면 어떨까요? 구현을 감추면 좀 더 가독성이 좋을 것 같아요!

Reply:
**데이터를 요청하지 말고 작업을 요청해라** 명심하겠습니다.